src/test/java/com/imagination/cbs/service/helper/BookingHrApproveHelperTest.java

```diff
@@ -0,0 +1,127 @@
+import com.imagination.cbs.service.AdobeSignService;
+import com.imagination.cbs.service.Html2PdfService;
+import com.imagination.cbs.service.LoggedInUserService;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BookingHrApproveHelperTest {
+
+	@InjectMocks
+	private BookingHrApproveHelper bookingHrApproveHelper;
+	
+	@Mock
+	private LoggedInUserService loggedInUserService;
+
+	@Mock
+	private Html2PdfService html2PdfService;
+
+	@Mock
+	private AdobeSignService adobeSignService;
+
+	@Mock
+	private BookingSaveHelper bookingSaveHelper;
+
+	@Mock
+	private EmailHelper emailHelper;
+	
+	@Test
+	public void shouldPrepareMailAndSendToHrWithBookingRevisionDetailsWhenUserIsAuthorized() throws IOException {
+		Booking booking = createBooking();
+		BookingRevision bookingRevision = createBookingRevision();
+		byte b[] = {20,10,30,5};
+		ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+		byteArrayOutputStream.write(b);
+		CBSUser cbsUser = createCBSUser();
+		
+		when(loggedInUserService.isCurrentUserInHRRole()).thenReturn(true);
+		when(bookingSaveHelper.getLatestRevision(booking)).thenReturn(bookingRevision);
+		when(html2PdfService.generateAgreementPdf(bookingRevision)).thenReturn(byteArrayOutputStream);
+		when(adobeSignService.uploadAndCreateAgreement((Mockito.any(InputStream.class)),Mockito.eq("service.pdf"))).thenReturn("CBS-111");
+		when(adobeSignService.sendAgreement("CBS-111", bookingRevision)).thenReturn("C-546");
+		when(loggedInUserService.getLoggedInUserDetails()).thenReturn(cbsUser);
+		when(bookingSaveHelper.saveBooking(booking, bookingRevision, 1006L,cbsUser)).thenReturn(booking);
+		
+		bookingHrApproveHelper.hrApprove(booking);
+		
+		verify(loggedInUserService).isCurrentUserInHRRole();
+		verify(bookingSaveHelper).getLatestRevision(booking);
+		verify(html2PdfService).generateAgreementPdf(bookingRevision);		
+		verify(emailHelper).prepareMailAndSendToHR(bookingRevision);
+		verify(adobeSignService).uploadAndCreateAgreement(Mockito.any(InputStream.class),Mockito.eq("service.pdf"));
+		verify(loggedInUserService).getLoggedInUserDetails();
+		verify(bookingSaveHelper).saveBooking(booking, bookingRevision, 1006L,cbsUser);
+	}
+
+	@Test(expected = CBSUnAuthorizedException.class )
+	public void shouldThrowCBSUnAuthorizedExceptionWhenUserIsUnAuthorized() {
+		
+		when(loggedInUserService.isCurrentUserInHRRole()).thenReturn(false);
+		
+		bookingHrApproveHelper.hrApprove(createBooking());
+		
+	}
+	
+	@Test(expected = CBSApplicationException.class )
+	public void shouldThrowCBSApplicationExceptionWhenBookingAlreadyApprovedOrNotInApprovalStatus() {
+		
+		ApprovalStatusDm approvalStatus = createApprovalStatusDm();
+		approvalStatus.setApprovalStatusId(1111L);
+		Booking booking = createBooking();
+		booking.setApprovalStatus(approvalStatus);
+		
+		when(loggedInUserService.isCurrentUserInHRRole()).thenReturn(true);
+		
+		bookingHrApproveHelper.hrApprove(booking);
+	}
+	private Booking createBooking() {
```

Comment:
can we move this method to some common place, from where we can use this domain creation in other test cases too.